service/controller/resource/promxy/servergroup/create.go

```diff
@@ -0,0 +1,61 @@
+package servergroup
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/prometheus-meta-operator/service/controller/resource/promxy"
+	"github.com/giantswarm/prometheus-meta-operator/service/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cluster, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	configMapName := key.PromxyConfigMapName()
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checking if %s configmap already exists ", configMapName))
+	configmap, err := r.k8sClient.K8sClient().CoreV1().ConfigMaps(key.PromxyConfigMapNamespace()).Get(ctx, configMapName, metav1.GetOptions{})
+
+	if apierrors.IsNotFound(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("configmap %s does not exists", configMapName))
+		return nil
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	promxyConfiguration, err := promxy.Deserialize(configmap.Data["values"])
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	url, err := url.Parse("https://kubernetes.default:443")
+	if err != nil {
+		return microerror.Mask(err)
+	}
```

Comment:
This can be moved inside `r.toServerGroup`.

Reply:
Sure, but technically, I think we can actually inject it from the kubernetes client when we start the operator. I just did not figure out how yet